src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectAssetFileWatcher.cs

```diff
@@ -159,6 +160,21 @@ protected override Task DisposeCoreAsync(bool initialized)
             return Task.CompletedTask;
         }
 
+        private static int? GetFileHashOrNull(string path)
+        {
+            int? hash = null;
+
+            try
+            {
+                hash = File.ReadAllText(path)?.GetHashCode();
```

Comment:
I would still not use this as the hashing approach - it requires allocating a giant string, doing unicode decoding, etc.  Using a decent hash that is built off a stream for the file contents would be much better.

Reply:
Which algorithm do you recommend? SHA256? Collisions are not too much of a concern, as this is not for security. But performance also isn't a concern since this is already a delayed task and in the background.